### ISSUE TITLE
[BUG-REPORT] `lstrip` stripping uncorrectly in some case

### ISSUE THREAD

**Issue by yohplala** [vaexio/vaex#1945]
**Description**
It seems like under some conditions, `lstrip` works uncorrectly.

This case does not work.
```python
import vaex

test_string = '/dai2/strades-2021-08.7768234.arrow'
header = test_string.rsplit('-',2)[0] + '-'    #   '/dai2/strades-'
vdf = vaex.from_arrays(fnames=[test_string])
vdf["fnames"].str.lstrip(header)
Out[1]: 
Expression = str_lstrip(fnames, '/dai2/strades-')
Length: 1 dtype: string (expression)
------------------------------------
0  021-08.7768234.arrow
```
Expected output is `2021-08.7768234.arrow`  (notice the `2` missing, and actually there is a `2` within the path).
Now, using a path without `2` (initially `dai2`, now `dai`), it works as expected.
```python
import vaex

test_string = '/dai/strades-2021-08.7768234.arrow'    # here, 'dai' instead of 'dai2'
header = test_string.rsplit('-',2)[0]  + '-'    #   '/dai/strades-'
vdf = vaex.from_arrays(fnames=[test_string])
vdf["fnames"].str.lstrip(header)
Out[5]: 
Expression = str_lstrip(fnames, '/dai/strades-')
Length: 1 dtype: string (expression)
------------------------------------
0  2021-08.7768234.arrow
```

**Software information**
 - Vaex version (`import vaex; vaex.__version__)`:
 {'vaex': '4.7.0',
 'vaex-core': '4.7.0.post1',
 'vaex-viz': '0.5.1',
 'vaex-hdf5': '0.11.1',
 'vaex-server': '0.8.1',
 'vaex-astro': '0.9.0',
 'vaex-jupyter': '0.7.0',
 'vaex-ml': '0.16.0',
 'vaex-graphql': '0.2.0'}
 - Vaex was installed via: pip / conda-forge / from source: mamba
 - OS: Ubuntu 20.04

**Additional information**
I am currently unable to install vaex-core 4.8.0 with mamba. Weirdly, if I do `mamba install vaex-core=4.8.0 -c conda-forge`, it will install a previous `vaex-server` (an older) version. If I then re-install last `vaex-server` with mamba, it will revert `vaex-core` to `4.7.0.post1`.

**Comment by yohplala** [comment]
Ok, it seems I have a workaround, but still, there does be a bug.

Thing is in the next step, with `str.slice`, I am cherry picking the date. "Full" code actually reads:
```python
vdf["fnames"] = vdf["fnames"].str.lstrip(header).str.slice(start=0, stop=7)
```
I could see that if I leave the `-` before the date, `str.lstrip` works correctly this time.
So, I am leaving the `-`, and will start the slicing 1 character later.
```python
import vaex

test_string = '/dai2/strades-2021-08.7768234.arrow'
header = test_string.rsplit('-',2)[0]    #   '/dai2/strades'
vdf = vaex.from_arrays(fnames=[test_string])
vdf["fnames"].str.lstrip(header).str.slice(start=1, stop=8)
Out[4]: 
Expression = str_slice(str_lstrip(fnames, '/dai2/strades'), start=1, s...
Length: 1 dtype: string (expression)
------------------------------------
0  2021-08
```
Yes! :)

**Comment by JovanVeljanoski** [comment]
Replying to the original post:

At first glance this does look like a bug. But I just tried it out, and the behavior of a vaex dataframe (the exampe you posted above) is consistent with that of a pandas dataframe for the same example, AND with pure python, like:

```python
'/dai2/strades-2021-08.7768234.arrow'.lstrip('/dai2/strades-')
 # returns '021-08.7768234.arrow'
```

So either there is a bug in.. python (??) or there is something about `lstrip` we don't understand. So maybe someone with more knowledge should chime in why this is the case. 

(I am glad you find a way to move forward however)

**Comment by yohplala** [comment]
> (I am glad you find a way to move forward however)

Alas!
The workaround does not work. There does be a weird behavior with `lstrip`.
The workaround works for the case above, but fails for another unit test.

```python
import vaex
test_string = "asset-strades-2021-09.1236.arrow"
header = test_string.rsplit('-',2)[0]    # here, I do not include the '-' : as per above workaround proposal
vdf = vaex.from_arrays(fnames=[test_string])
vdf["fnames"].str.lstrip(header)        # expecting to have the resulting string starting with '-'
Out[9]: 
Expression = str_lstrip(fnames, 'asset-strades')
Length: 1 dtype: string (expression)
------------------------------------
0  2021-09.1236.arrow
```
Correct result should then include the `-`.
```python
-2021-09.1236.arrow
```

Thanks @JovanVeljanoski for your feedback!
I delve a bit more in the tpic, and this [SO question/answer](https://stackoverflow.com/questions/16891340/remove-a-prefix-from-a-string) give the picture.

Basically, `lstrip`continues to remove characters at the start of the string (any combination of them) that are specified by provided characters. (`header`).
So in above case, `2` is a character to remove, so `lstrip` removes it.
__`lstrip` stop removing only when it encounters a character not in input list of characters.__

I am closing the ticket then.
Have a good day and thanks again!
Bests,

**Comment by JovanVeljanoski** [comment]
Hey. Glad you figured it out. 
How about this? Or something similar?
`df.fnames.str.replace(pat=header, repl='')`

**Comment by yohplala** [comment]
> Hey. Glad you figured it out. How about this? Or something similar? `df.fnames.str.replace(pat=header, repl='')`

Dear Jovan, thank you very much for your continued support!
Well, during the day, I thought that I would take more out of `str.slice()` as I am already using it to get the date.
So I went to a
```python
slice_start = len(header)+1
vdf["fnames"].str.slice(start=slice_start, stop=slice_start+7)
```
And now, all my test cases are ok! Great! :)
(I removed everywhere all `lstrip` and `rstrip` I was using as the behavior in case of random file paths is much too uncontrolled)

Thanks again!
Bests,
